tests/unit/test_main.py

```diff
@@ -209,14 +222,21 @@ def test_noviaction_experimenter_pack_unpack(self):
                 assert unpacked.reserved == 0
                 assert unpacked.novi_action_type.value == action_type_val.value
 
+                assert isinstance(novi_class.from_of_action(action), novi_class)
+                as_of_action = novi_class().as_of_action()
+                assert isinstance(as_of_action, action_class)
+                assert as_of_action.customer == 0xFF
+                assert as_of_action.reserved == 0
+                assert as_of_action.novi_action_type.value == action_type_val.value
+
     def test_noviaction_set_bfd_data(self):
         """Test NoviActionSetBfdData experimenter pack and unpack."""
 
-        port_no = 2
-        my_disc = 1
-        interval = 5
-        multiplier = 3
-        keep_alive_timeout = 15
+        port_no = UBInt32(2)
+        my_disc = UBInt32(1)
+        interval = UBInt32(5)
+        multiplier = UBInt8(3)
+        keep_alive_timeout = UBInt8(15)
```

Comment:
Just one question please, @viniarck: why not instantiate the parameters with the proper UBIntXX as part of the `NoviActionSetBfdData` class instead of in the client? What if the client just instantiate this class using int values (which is more common):
```
        action = OFNoviActionSetBfdData(
             port_no=2,
            my_disc=1,
            interval=5,
            multiplier=3,
            keep_alive_timeout=15,
        )
```

Reply:
@italovalcy indeed, this could blow up, from what I checked, where this is this is not being called and passing `UBInt*` values, however I initially parametrized them based on how the `from_of_action` was assuming that they'd be `UBInt`. I believe it's safe to do a int cast here [since `UBIntBase` has this compatible construtor](https://github.com/kytos-ng/python-openflow/blob/master/pyof/foundation/base.py#L265-L269) to support both cases. Please this commit here https://github.com/kytos-ng/noviflow/pull/8/commits/63afc08bf95d65218a3db754d1b37e18b0a1bc7f, let me know what you think. 

Also, comparing with some usages from other classes on `pyof` I believe that indeed having a `int` after unpacking is what's likely to be expected: 

```
In [1]: from pyof.foundation.network_types import (
   ...:     ARP, VLAN, Ethernet, GenericTLV, IPv4, IPv6)

In [2]:         raw = b'FP\x00$\x00\x00\x00\x00\x80\x06W'
   ...:         raw += b'\xf4\n\x9aN\x81\xc0\xa8\xc7\xcc1000somemoredata'
   ...:         expected = IPv4(dscp=20, ttl=128, protocol=6, source="10.154.78.129",
   ...:                         destination="192.168.199.204", options=b'1000',
   ...:                         data=b'somemoredata')
   ...:         expected.pack()
   ...:         unpacked = IPv4()
   ...:         unpacked.unpack(raw)

In [3]: type(unpacked.protocol)
Out[3]: int

In [4]: unpacked.protocol
Out[4]: 6

In [5]:     protocol = UBInt8()
```

I've also noticed that on `of_core` we have similar classes using the `from_of_action` assuming it'd be an `UBInt` like this one on [`ActionOuput`](https://github.com/kytos-ng/of_core/blob/master/v0x04/flow.py#L78-L80), maybe we should also map to make sure it's safe there as well? cc'ing @ajoaoff for his info and feedback as well when he's back from vacation. We don't have much test coverage for `from_of_action` usages yet from what I've found in the codebase.

Reply:
Sounds good to me! Regarding the question on `of_core`, I'll leave it to @ajoaoff answer.

Reply:
> Just one question please, @viniarck: why not instantiate the parameters with the proper UBIntXX as part of the `NoviActionSetBfdData` class instead of in the client? What if the client just instantiate this class using int values (which is more common):
> 
> ```
>         action = OFNoviActionSetBfdData(
>              port_no=2,
>             my_disc=1,
>             interval=5,
>             multiplier=3,
>             keep_alive_timeout=15,
>         )
> ```

@italovalcy I don't know of a case where a client instatiates theses classes directly. They are created either from a `of_core` action class `as_of_action` or from unpacking.

Reply:
> @italovalcy indeed, this could blow up, from what I checked, where this is this is not being called and passing `UBInt*` values, however I initially parametrized them based on how the `from_of_action` was assuming that they'd be `UBInt`. I believe it's safe to do a int cast here [since `UBIntBase` has this compatible construtor](https://github.com/kytos-ng/python-openflow/blob/master/pyof/foundation/base.py#L265-L269) to support both cases. Please check this commit here [63afc08](https://github.com/kytos-ng/noviflow/commit/63afc08bf95d65218a3db754d1b37e18b0a1bc7f), let me know what you think.
> 
> Also, comparing with some usages from other classes on `pyof` I believe that indeed having a `int` after unpacking is what's to be expected:
> 
> ```
> In [1]: from pyof.foundation.network_types import (
>    ...:     ARP, VLAN, Ethernet, GenericTLV, IPv4, IPv6)
> 
> In [2]:         raw = b'FP\x00$\x00\x00\x00\x00\x80\x06W'
>    ...:         raw += b'\xf4\n\x9aN\x81\xc0\xa8\xc7\xcc1000somemoredata'
>    ...:         expected = IPv4(dscp=20, ttl=128, protocol=6, source="10.154.78.129",
>    ...:                         destination="192.168.199.204", options=b'1000',
>    ...:                         data=b'somemoredata')
>    ...:         expected.pack()
>    ...:         unpacked = IPv4()
>    ...:         unpacked.unpack(raw)
> 
> In [3]: type(unpacked.protocol)
> Out[3]: int
> 
> In [4]: unpacked.protocol
> Out[4]: 6
> 
> In [5]:     protocol = UBInt8()
> ```
> 
> I've also noticed that on `of_core` we have similar classes using the `from_of_action` assuming it'd be an `UBInt` like this one on [`ActionOuput`](https://github.com/kytos-ng/of_core/blob/master/v0x04/flow.py#L78-L80), maybe we should also map to make sure it's safe there as well? cc'ing @ajoaoff for his info and feedback as well when he's back from vacation. We don't have much test coverage for `from_of_action` usages yet from what I've found in the codebase.

Although `IPv4` is returning `int` after unpacking, it does not happen with the actions. If you make the same test with, for example, `pyof.v0x04.common.action.ActionOutput` after unpacking the `port` is a `UBInt32`. In fact, if you call `from_of_action` passing an OF action created with `int`s, it will fail. I don't think it ever happens in real cases, OF actions are created either from `of_core` action classes or from unpacking. I'm not sure if there is any problem having an `int` instead of a `UBInt`, but I think we should keep `from_of_action` as it is, assuming it is a `UBInt`.